examples/robot_localization_failure/run.py

```diff
@@ -1,103 +1,311 @@
+            train_datasets.append(train_images)
+            test_images = FeatureImagesData(
+                test.data.collect(),
+                image_size=config.architecture.image_size,
+                width_meters=config.architecture.width_meters,
+            )
+            test_datasets.append(test_images)
+        except Exception as e:
+            msg = f"Error loading {parquet_path}: {e}"
+            logger.exception(msg)
+            continue
+
+    # Combinedatasets
+    train_data = pl.concat([train.data for train in train_datasets])
+    test_data = pl.concat([test.data for test in test_datasets])
+
+    # Debug: Print columns to verify structure
+    logger.info(f"Train data columns: {train_data.columns}")
+    logger.info(f"Test data columns: {test_data.columns}")
+
+    # Create and train the learner
+    learner = ImageBasedLightningLearner(
+        module=CNN(
+            image_size=config.architecture.image_size,
+            in_channels=3,
+            learning_rate=config.learning.learning_rate,
+        ),
+        batch_size=config.learning.batch_size,
+        max_epochs=config.learning.epochs,
+        dataset_factory=lambda data: FeatureImagesData(
+            data,
+            image_size=config.architecture.image_size,
+            width_meters=config.architecture.width_meters,
+        ),
+        dataset_kwargs={
+            "image_size": config.architecture.image_size,
+            "width_meters": config.architecture.width_meters,
+        },
     )
-    | Drop("/scan", "/particle_cloud", "/momo/pose", "/amcl_pose")
-    # detect experiment slice points based on delocalization events
-    | DetectDelocalizations("/delocalizations", name="slice_points")
-    | Drop("/delocalizations")
-    | SliceTimeSeries(
-        time_series="measurements",
-        slice_points="slice_points",
+    model = learner.learn(
+        inputs=train_data.drop(["is_delocalized"]),
+        outputs=train_data.select(["is_delocalized"]),
     )
-    | Drop("slice_points")
-    # detect localization status based on position and heading errors
-    | LocalizationStatus(
-        time_series="measurements",
-        ground_truth="/momo/pose",
-        estimation="/amcl_pose",
-        position_threshold=config.localization.position_threshold,
-        heading_threshold=config.localization.heading_threshold,
+
+    # Evaluate the model
+    metrics = [
+        Accuracy(),
+        ClassificationReport(),
+        FBetaScore(beta=1.0),
+        PrecisionScore(),
+        Recall(),
+    ]
+    report = evaluate_offline(
+        model,
+        DataFrame(test_data),
+        inputs=test_data.drop(["is_delocalized"]).columns,
+        outputs=["is_delocalized"],
+        metrics=metrics,
     )
-)
+    print(report)
+    # Save the model
+    out_path = f"models/{rosbag_dir.name}.pt"
+    msg = f"Saving model to {out_path}"
+    logger.info(msg)
+    torch.save(model.module.state_dict(), out_path)
+
+
+def get_rosbag_paths(
```

Comment:
if you specify the rosbag paths in the `config.yaml` as a list, you can probably remove the entire function.